grails-app/controllers/org/broadinstitute/orsp/api/ReportController.groovy

```diff
@@ -21,24 +24,54 @@ class ReportController extends AuthenticatedController {
         render([error: "Not Implemented"] as JSON)
     }
 
-    def funding() {
+    def getFunding() {
+        Long i = 0
+        JSON.registerObjectMarshaller(Funding) {
+            def output = [:]
+            output['id'] = i++
+            output['type'] = it.issue.type
+            output['projectKey'] = it.issue.projectKey
+            output['protocol'] = it.issue.protocol
+            output['summary'] = it.issue.summary
+            output['status'] = it.issue.status
+            output['source'] = it.source
+            output['name'] = it.name
+            output['awardNumber'] = it.awardNumber
+            output['pis'] = getPIsDisplayName((Issue) it.issue)//it.issue.getPIs().collect{r -> getUser(r).displayName}
+            return output
+        }
+
         PaginationParams pagination = new PaginationParams(
-                draw: params.getInt("draw")?: 1,
+                draw: params.getInt("draw") ?: 1,
                 start: params.getInt("start")?: 0,
                 length: params.getInt("length")?: 10,
-                orderColumn: params.getInt("order[0][column]")?: 0,
-                sortDirection: params.get("order[0][dir]")? params.get("order[0][dir]").toString() : "asc",
-                searchValue: params.get("search[value]")? params.get("search[value]").toString() : null)
+                orderColumn: params.getInt("orderColumn")? params.getInt("orderColumn"): 1,
+                sortDirection: params.get("sortDirection")? params.get("sortDirection").toString() : "desc",
+                searchValue: params.get("searchValue")? params.get("searchValue").toString() : null)
         render(queryService.queryFundingReport(pagination) as JSON)
     }
 
+    def getAllFundings() {
+        Collection<Funding> fundings = queryService.getAllFundings()
+        render(fundings as JSON)
+    }
+
     def aahrppMetrics() {
         Collection<AAHRPPMetrics> metrics = queryService.findAllAAHRPPMetrics()
         String content = reportService.createReport(metrics.toList())
         response.setHeader("Content-disposition", "attachment; filename=AAHRPPMetrics.csv")
         response.outputStream << content
     }
 
+    static List<String> getPIsDisplayName(Issue issue) {
+        List<String> piUserNames = issue*.getPIs().flatten().unique()
```

Comment:
Review if * (spread) is required here ...

Reply:
Yeah, I don't think it's needed. This could probably just be 
```suggestion
        List<String> piUserNames = issue?.getPIs()?.flatten()?.unique()
```
If you have to return a real list, then pre-initialize it and add whatever comes back from the pis call.

Reply:
I should have kept reading ... this whole block can be simplified ... brb.

Reply:
Indeed, spread operator was not required here. On the other hand I simplified this method as suggested using null operators.